src/datastation/scripts/open_access_archeodepot.py

```diff
@@ -0,0 +1,150 @@
+        for row in csv_reader:
+            key = to_key(row["rest"][-1].strip())
+            uri = rights_holder_to_license_uri.get(key, "")
+            if uri:
+                doi_to_license_uri[row["DOI"]] = uri
+            else:
+                logging.warning("no license for line {}: {}".format(csv_reader.line_num, row))
+    return doi_to_license_uri
+
+
+def read_doi_to_dag_raporten(dag_raporten_file):
+    doi_to_dag_raporten = {}
+    with open(dag_raporten_file, "r") as input_file_handler:
+        csv_reader = csv.DictReader(input_file_handler, delimiter=',',
+                                    fieldnames=["dataset_id", "DOI"], restkey="files")
+        next(csv_reader)
+        for row in csv_reader:
+            doi_to_dag_raporten[to_key(row["DOI"])] = list(filter(lambda item: item != "", row["files"]))
+    return doi_to_dag_raporten
+
+
+def read_rights_holder_to_license(licenses_file):
+    rights_holder_to_license_uri = {}
+    with open(licenses_file, "r") as input_file_handler:
+        csv_reader = csv.DictReader(input_file_handler, delimiter=',')
+        for row in csv_reader:
+            rights_holder_to_license_uri[to_key(row["RIGHTS_HOLDER"])] = row["URI"]
+    return rights_holder_to_license_uri
+
+
+def to_key(name):
+    return re.sub("[^a-zA-Z0-1]", "_", name)
+
+
+def update(doi, uri, dag_raporten, server_url, api_token, dry_run):
+    resp_data = get_dataset_metadata(server_url, api_token, doi)
+    dirty = False
+    if resp_data['license']['uri'] != uri:
+        dirty = True
+        json_data = json.dumps({"http://schema.org/license": uri})
+        logging.info(json_data)
+        if not dry_run:
+            replace_dataset_metadata(server_url, api_token, doi, json_data)
+    accessible_dag_raporten = list(filter(
+        lambda file: not file['restricted'] and file_path(file) in dag_raporten,
+        resp_data['files']))
+    restricted_other_files = list(filter(
+        lambda file: file['restricted'] and file_path(file) not in dag_raporten,
+        resp_data['files']))
+    logging.info("number of: dag_raporten={}, accessible_dag_raporten={}, restricted_other_files={}; {}".format(
+        len(dag_raporten), len(accessible_dag_raporten), len(restricted_other_files), dag_raporten))
+    has_accessible_dag_raporten = len(accessible_dag_raporten) > 0
+    has_dag_raporten = len(dag_raporten) > 0
+    if bool(resp_data['fileAccessRequest']) != has_dag_raporten:
+        logging.info("(re)setting access request {}".format(has_accessible_dag_raporten))
+        dirty = True
+        if not dry_run:
+            change_access_request(server_url, api_token, doi, has_dag_raporten)
+    change_files(False, restricted_other_files, server_url, api_token, dry_run)
+    if has_accessible_dag_raporten:
+        if not resp_data.get("termsOfAccess", None):
+            logging.warning("no terms of access, can't restrict dag-raporten of {}".format(doi))
```

Comment:
let's add a `TermsOfAccess` of `N/a` if none is present. and still restrict them. otherwise we will end up in a wrong situation, with the dataset opened, and the files not closed.